### PR TITLE
feature/s21_strlen

### LINKED ISSUE #28: `s21_strlen` - Computes the length of str (excluding \0).
## Description:  
The task is to implement the `s21_strlen` function, which computes the length of the string `str` up to but not including the terminating null character.  

## Requirements:  

- [x] The function must be named `s21_strlen` and conform to the C11 standard.  
- [x] The implementation must be placed in the `s21_string.h` library.  
- [x] Follow the Google C++ Style Guide for code formatting.  
- [x] Ensure the function works with z-strings made of single-byte characters in ASCII encoding.  
- [x] Write unit tests using the Check library to verify the correctness of the implementation. The tests must cover at least 80% of the function (checked using gcov).  
- [x] The function must not use any standard library functions for string length calculation (e.g., `strlen`). 

## Expected Behavior:  
- For input `"Hello"`, the function should return `5`.  
- For input `""` (empty string), the function should return `0`.  
- For input `NULL`, the behavior should match the standard `strlen` function (undefined, but tests should handle it gracefully).  

## Additional Notes:  
- Ensure the implementation is efficient and avoids unnecessary loops or operations.  
- Include edge cases in unit tests (e.g., strings with maximum length, special characters).

### PR DESCRIPTION
**Pull request by DarkDomian**
## implementation of `strlen`

- just count the number of bytes in sting and return it
- watch on test-cases, probably they want fix

### CODE REVIEW

File: src/s21_strlen.c [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#include "../headers/s21_string.h"
+
+s21_size_t s21_strlen(const char *str) {
+    s21_size_t i;
+    for (i = 0; str[i] != '\0'; ++i);
```
Comment (ParthenGithub) [inline src/s21_strlen.c]: Any specific reason to use ++i and not i++?

File: tests/test_strlen.c [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+// Test case 5: Long string
+START_TEST(test_strlen_long_string) {
+    const char *str = "This is a very long string to test if s21_strlen handles long inputs correctly.";
+    ck_assert_int_eq(s21_strlen(str), strlen(str));
+}
+END_TEST
+
+// Test case 6: String with null terminator in the middle (undefined behavior, but should not crash)
+START_TEST(test_strlen_null_middle) {
+    const char *str = "Hello\0 World";
```
Comment (ParthenGithub) [inline tests/test_strlen.c]: According to C11 standard this is valid "Hello" string. 

_C11, Section 7.1.1, "Definitions of terms"):_

> "string is a contiguous sequence of characters terminated by and including the _**first null**_ character."

**Review by ParthenGithub — APPROVED** [review]
Well done

File: src/s21_strlen.c (hunk 1 above)
Comment (DarkDomian) [reply src/s21_strlen.c]: just save the processor's time (optimizing compiler will do the same btw)

for more information - open **godbolt** and make two different loop with `++i` and `i++` and have a look to the assembler

File: tests/test_strlen.c (hunk 2 above)
Comment (DarkDomian) [reply tests/test_strlen.c]: We can make for testing list of strings, like {"sting", "another one", "and another"}
but in the memory (depends on how the memory was allocated) they are the same.

I just drop down the boring process of creating list of strings for one case

Comment (ParthenGithub) [reply tests/test_strlen.c]: Don't bother. We don't really have a time for that :)

File: tests/test_strlen.c [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    ck_assert_int_eq(s21_strlen(str), strlen(str));
+}
+END_TEST
+
+// Test case 7: NULL input (undefined behavior, but should not crash)
+START_TEST(test_strlen_null_input) {
+    const char *str = NULL;
+    if (str) {
```
Comment (gashturik) [inline tests/test_strlen.c]: cppcheck gave message `  tests/test_strlen.c:50:9: style: Condition 'str' is always false [knownConditionTrueFalse]`

**Review by gashturik — COMMENTED** [review]
**Wrong style of code!!!!**

**Comment by DarkDomian** [comment]
Approve merge please. I fix the problems

**Comment by Amfir** [comment]
Are you planning to fix problems in develop? @DarkDomian

**Review by gashturik — APPROVED** [review]

**Comment by ParthenGithub** [comment]
```c
tests/test_strlen.c:50:9: style: Condition 'str' is always false [knownConditionTrueFalse]
    if (str) {
        ^
tests/test_strlen.c:49:23: note: Assignment 'str=NULL', assigned value is 0
    const char *str = NULL;
                      ^
tests/test_strlen.c:50:9: note: Condition 'str' is always false
    if (str) {
```

@gashturik why it's merged?
